### PR TITLE
Added 105Report section 2.9

### PR DESCRIPTION
**Pull request by moshonk**
@ssmusoke 
Added mappings for 105report section 2.9 
cc @ningosi

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/ugandaemrreports/library/Moh105CohortLibrary.java [hunk 1]
```diff
@@ -30,9 +33,6 @@
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
 
-import java.util.Arrays;
-import java.util.Date;
-
 /**
```
Comment (ssmusoke) [inline api/src/main/java/org/openmrs/module/ugandaemrreports/library/Moh105CohortLibrary.java]: @moshonk @ningosi Please remove created by from all files in reporting module - that can be a separate PR

**Comment by ssmusoke** [comment]
@moshonk Why does this PR have changes to HMIS_105-2.1-2.7.xls?

File: api/src/main/java/org/openmrs/module/ugandaemrreports/library/Moh105CohortLibrary.java (hunk 1 above)
Comment (moshonk) [inline api/src/main/java/org/openmrs/module/ugandaemrreports/library/Moh105CohortLibrary.java]: @ssmusoke, Noted and done.

**Comment by ssmusoke** [comment]
@moshonk Please can you put the removal of created by in another PR as it will pollute this one

**Comment by ssmusoke** [comment]
@moshonk You will also need to add a link for the report in the ugandaemr_reports_app.json

**Comment by moshonk** [comment]
@ssmusoke, I have reverted changes made to the **HMIS_105-2.1-2.7.xls** file.  I have also undone 'the removal of Created By' action done on this PR

File: api/src/main/java/org/openmrs/module/ugandaemrreports/library/Moh105CohortLibrary.java [hunk 2]
```diff
@@ -211,4 +211,45 @@ public CohortDefinition hasObsAndEncounter(String encounterType, Concept q, Conc
         cd.setCompositionString("ancEncounter AND hasObs");
         return cd;
     }
+
+	/**
+	 * Number Tetanus Immunizations done
+	 * 
+	 * @return CohortIndicator
+	 */
+	public CohortDefinition tetanusImmunizationsDone(int doseNumber) {
```
Comment (ssmusoke) [inline api/src/main/java/org/openmrs/module/ugandaemrreports/library/Moh105CohortLibrary.java]: @moshonk Can you limit this Cohort to the ANC encounter - as this is the only one for pregnant women, and also add the definition for non-pregnant women for future forms... You could just pass a parameter for pregnant

Comment (moshonk) [inline api/src/main/java/org/openmrs/module/ugandaemrreports/library/Moh105CohortLibrary.java]: @ssmusoke.This has been done. Also added the link to the report in the ugandaemr_reports_app.json file

File: api/src/main/java/org/openmrs/module/ugandaemrreports/library/Moh105IndicatorLibrary.java [hunk 3]
```diff
@@ -342,9 +342,9 @@ public CohortIndicator visitsForWomenInFirstTrimester(){
 	 * 
 	 * @return CohortIndicator
 	 */
-	public CohortIndicator tetanusImmunizationsDone(int doseNumber) {
+	public CohortIndicator tetanusImmunizationsDone(int doseNumber, Boolean pregnant) {
```
Comment (ssmusoke) [inline api/src/main/java/org/openmrs/module/ugandaemrreports/library/Moh105IndicatorLibrary.java]: @moshonk Why not use the primitive boolean here instead of the class Boolean

File: api/src/main/java/org/openmrs/module/ugandaemrreports/reports/SetupMoH105_28_To_212ReportBuilder.java [hunk 4]
```diff
@@ -116,11 +116,11 @@ protected DataSetDefinition tetanusImmunizations() {
         
         String params = "startDate=${startDate},endDate=${endDate}";
         
-        dsd.addColumn("T1PS","T1: T1-Dose 1", ReportUtils.map(indicatorLibrary.tetanusImmunizationsDone(1),params), "");
-        dsd.addColumn("T2PS","T2-Dose 2", ReportUtils.map(indicatorLibrary.tetanusImmunizationsDone(2),params), "");
-        dsd.addColumn("T3PS","T3-Dose 3", ReportUtils.map(indicatorLibrary.tetanusImmunizationsDone(3), params), "");
-        dsd.addColumn("T4PS","T4-Dose 4", ReportUtils.map(indicatorLibrary.tetanusImmunizationsDone(4), params), "");
-        dsd.addColumn("T5PS","T5-Dose 5", ReportUtils.map(indicatorLibrary.tetanusImmunizationsDone(5), params), "");
+        dsd.addColumn("T1PS","T1: T1-Dose 1", ReportUtils.map(indicatorLibrary.tetanusImmunizationsDone(1, true),params), "");
```
Comment (ssmusoke) [inline api/src/main/java/org/openmrs/module/ugandaemrreports/reports/SetupMoH105_28_To_212ReportBuilder.java]: @moshonk As discussed need to add the TT dose computation for the non-pregnant

Comment (moshonk) [inline api/src/main/java/org/openmrs/module/ugandaemrreports/reports/SetupMoH105_28_To_212ReportBuilder.java]: @ssmusoke. This has been done

File: api/src/main/java/org/openmrs/module/ugandaemrreports/library/Moh105IndicatorLibrary.java (hunk 3 above)
Comment (moshonk) [inline api/src/main/java/org/openmrs/module/ugandaemrreports/library/Moh105IndicatorLibrary.java]: @ssmusoke, Noted and changed to a primitive boolean

File: api/src/main/java/org/openmrs/module/ugandaemrreports/reporting/metadata/Metadata.java [hunk 5]
```diff
@@ -38,6 +38,14 @@
         public static final String TB_TREATMENT_START_DATE = "dce02eca-30ab-102d-86b0-7a5022ba4115";
         public static final String ON_TB_TREATMENT = "dcdaa6b4-30ab-102d-86b0-7a5022ba4115";
         public static final String ASSESSED_FOR_MALNUTRITION = "dc655734-30ab-102d-86b0-7a5022ba4115";
+
+        public static final String TETANUS_DOSE_GIVEN = "39217e3d-6a39-4679-bf56-f0954a7ffdb8";
+		public static final String FIRST_DOSE = "0192ca59-b647-4f88-b07e-8fda991ba6d6";
```
Comment (ssmusoke) [inline api/src/main/java/org/openmrs/module/ugandaemrreports/reporting/metadata/Metadata.java]: @moshonk prefix the doses with TETANUS_ so that this does not cause issues with other drugs given in multiple does

Comment (moshonk) [inline api/src/main/java/org/openmrs/module/ugandaemrreports/reporting/metadata/Metadata.java]: @ssmusoke, the dosage concepts are generic as defined in the concept dictionary and may be reused in the future for defining dosages for other drugs.. hence I don't think it's necessary to prefix them.

Comment (ssmusoke) [inline api/src/main/java/org/openmrs/module/ugandaemrreports/reporting/metadata/Metadata.java]: @moshonk Double checked and you are right
